sdk/python/feast/infra/online_stores/contrib/hazelcast_online_store/hazelcast_online_store.py

```diff
@@ -0,0 +1,252 @@
+                    progress(1)
+
+    def online_read(
+        self,
+        config: RepoConfig,
+        table: FeatureView,
+        entity_keys: List[EntityKeyProto],
+        requested_features: Optional[List[str]] = None,
+    ) -> List[Tuple[Optional[datetime], Optional[Dict[str, ValueProto]]]]:
+        online_store_config = config.online_store
+        assert isinstance(online_store_config, HazelcastOnlineStoreConfig)
+
+        client = self._get_client(online_store_config)
+        entries: List[Tuple[Optional[datetime], Optional[Dict[str, ValueProto]]]] = []
+        fv_map = client.get_map(_table_id(config.project, table))
+
+        hz_keys = []
+        entity_keys_str = {}
+        for entity_key in entity_keys:
+            feature_keys = []
+            entity_key_str = str(
+                serialize_entity_key(
+                    entity_key,
+                    entity_key_serialization_version=config.entity_key_serialization_version,
+                ).hex()
+            )
+            if requested_features:
+                for feature in requested_features:
+                    feature_keys.append(entity_key_str + feature)
+            else:
+                for f in table.features:
+                    feature_keys.append(entity_key_str + f.name)
+            hz_keys.extend(feature_keys)
+            entity_keys_str[entity_key_str] = feature_keys
+
+        data = fv_map.get_all(hz_keys).result()
+        entities = []
+        for _key in hz_keys:
+            try:
+                data[_key] = data[_key].loads()
+                entities.append(data[_key]["entity_key"])
+            except KeyError:
+                continue
+
+        for curr_key in entity_keys_str:
+            if curr_key in entities:
+                entry = dict()
+                event_ts: Optional[datetime] = None
+                for _key in entity_keys_str[curr_key]:
+                    row = data[_key]
+                    value = ValueProto()
+                    value.ParseFromString(bytes.fromhex(row["feature_value"]))
+                    entry[row["feature_name"]] = value
+                    event_ts = datetime.utcfromtimestamp(row["event_ts"])
+                entries.append((event_ts, entry))
+            else:
+                entries.append((None, None))
+        return entries
+
+    def update(
+        self,
+        config: RepoConfig,
+        tables_to_delete: Sequence[FeatureView],
+        tables_to_keep: Sequence[FeatureView],
+        entities_to_delete: Sequence[Entity],
+        entities_to_keep: Sequence[Entity],
+        partial: bool,
+    ):
+        online_store_config = config.online_store
+        assert isinstance(online_store_config, HazelcastOnlineStoreConfig)
+
+        client = self._get_client(online_store_config)
+        project = config.project
+
+        for table in tables_to_keep:
+            client.sql.execute(
+                f"""CREATE OR REPLACE MAPPING {_table_id(project, table)} (
```

Comment:
Do we still need to create an SQL mapping? I thought you were using the map directly. Same comment for other SQL code.

Reply:
This is for creating the map schemas of feature views in the online store, so users can see it. We can pass this step since Hazelcast allow us but I wanted to add to provide some functionality in here.

Also it's easier to remove all entries from Hazelcast using SQL. Functions with SQL queries won't executed very frequently so I think it does not add any overhead.